### PR TITLE
Feature/#67edit a contact

### PR DESCRIPTION
**Pull request by TDManning**
## Type of Change

- [ ✅] feature ⛲
- [ ✅] documentation update 📃
- [ ✅] testing 🧪

## Description
The update contact endpoint `(PATCH /api/v1/users/:user_id/contacts/:id)` allows a user to update an existing contact's details. It checks if the contact exists, ensures the user has permission to update it, and applies the changes if valid. If the contact is not found, it returns a 404 error, and if the update fails due to validation issues, it returns a 422 error. 

## Motivation and Context
This endpoint needed to be created in the back end in order to enable a user to update an existing contact

## Related Tickets
Front End ticket to complete this endpoint

## Added Test?
- [✅ ] Yes 🫡
-Model and Request tests include: Happy Path, Sad Path and Edge cases.
- [✅ ] All previous tests still pass 🥳

## Checklist:

- [✅ ] My code follows the code style of this project.
- [ ✅] My change requires a change to the documentation.
- [ ✅] I have updated the documentation accordingly.
- [ ] This PR includes a Migration and I have notified the deployment team and PM so they can run the migration after the PR is merged.

### CODE REVIEW

File: app/controllers/api/v1/contacts_controller.rb [hunk 1]
```diff
@@ -47,6 +46,25 @@ def show
         end
       end
 
+      def update
```
Comment (Crosswolfv1) [inline app/controllers/api/v1/contacts_controller.rb:49]: Looks great, follows good controller rules

File: spec/requests/api/v1/contacts/update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        expect(response).to be_successful
+        json = JSON.parse(response.body, symbolize_names: true)[:data][:attributes]
+
+        expect(json[:email]).to eq("updated@example.com")
+        expect(json[:phone_number]).to eq("555-555-5556")
+        expect(json[:notes]).to eq("Updated notes")
+      end
+    end
+
+    context "Sad Paths" do
```
Comment (Crosswolfv1) [inline spec/requests/api/v1/contacts/update_spec.rb:33]: Covers all sad cases, and I cannot think of any edge cases for it to cover

**Review by Crosswolfv1 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -854,6 +854,80 @@ Successful Response:
...
+                "street_address": "123 Innovation Way",
+                "city": "San Francisco",
+                "state": "CA",
+                "zip_code": "94107",
+                "notes": "Reached out on LinkedIn, awaiting response."
+            }
+        }
+    }
+}
+
```
Comment (kaelinpsalazar) [inline README.md:930]: Good readme practice.

File: spec/requests/api/v1/contacts/update_spec.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        expect(response).to have_http_status(:unprocessable_entity)
+        json = JSON.parse(response.body, symbolize_names: true)
+      
+        expect(json[:message]).to include("Phone number must be in the format '555-555-5555'")
+        expect(json[:status]).to eq(422)
+      end
+      
+    end
+  end
+end
```
Comment (kaelinpsalazar) [inline spec/requests/api/v1/contacts/update_spec.rb:72]: I agree with jeremiah, your testing is solid and i cant think of a edgecase that would otherwise be needed.

**Review by kaelinpsalazar — CHANGES_REQUESTED** [review]
Just a small thing, but you should go below jim on the README contributors list

File: README.md [hunk 5]
```diff
@@ -1213,13 +1283,17 @@ Successful Response:
...
 
+**Manning, Terra**
+- [Github](https://github.com/TDManning/)
+- [LinkedIn](https://www.linkedin.com/in/terra-manning/)
+
 **Macur, Jim**
-   - [Github](https://github.com/jimmacur)
-   - [LinkedIn](https://www.linkedin.com/in/jimmacur/)
+- [Github](https://github.com/jimmacur)
+- [LinkedIn](https://www.linkedin.com/in/jimmacur/)
```
[the remark this answers is not in the document]
Comment (TDManning) [reply README.md:1293]: Good Catch!

**Review by kaelinpsalazar — APPROVED** [review]
Thanks for making the change! Approved!

File: README.md [hunk 6]
```diff
@@ -1214,12 +1284,16 @@ Successful Response:
...
+
+**Manning, Terra**
+- [Github](https://github.com/TDManning/)
+- [LinkedIn](https://www.linkedin.com/in/terra-manning/)
 
 **Messersmith, Renee**
-   - [Github](https://github.com/reneemes)
-   - [LinkedIn](https://www.linkedin.com/in/reneehessersmith/)
+- [Github](https://github.com/reneemes)
+- [LinkedIn](https://www.linkedin.com/in/reneemessersmith/)
```
Comment (epintozzi) [inline README.md:1296]: Oh nice catch on the typo. Thanks for fixing!
